### PR TITLE
Clean up PSK provisions

### LINKED ISSUE #666: Clean up PSK specification
* `psk_nonce` currently described at time of use.  In the "Pre-Shared Keys" section, we should just specify that it must always be a fresh value of length `KDF.Nh`
* The ReInit provisions require `epoch=1`, but those for branching do not.  Presumably these should be the same?
* Currently, the processing rules for PreSharedKey proposal says “MUST be external”.  This seems overly restrictive; for example, it rules out the scenario in Figure 8.  Suggest “MUST NOT be branch/reinit” instead.

### PR DESCRIPTION
**Pull request by bifurcation**
Closes #666 

Also covers the PSK part of #673

### CODE REVIEW

File: draft-ietf-mls-protocol.md [hunk 1]
```diff
@@ -3366,8 +3363,8 @@ struct {
 } PreSharedKey;
 ~~~
 
-The `psktype` of the pre-shared key MUST be `external` and the `psk_nonce` MUST
-be a randomly sampled nonce of length `KDF.Nh`. When processing a Commit message
+A PreSharedKey proposal MUST NOT contain a PSK of type `resumption` and usage
```
Comment (Bren2010) [inline draft-ietf-mls-protocol.md:3366]:
```suggestion
A PreSharedKey proposal MUST NOT contain a PSK of type `resumption` that has usage
```

**Review by kkohbrok — APPROVED** [review]

File: draft-ietf-mls-protocol.md [hunk 2]
```diff
@@ -3178,13 +3180,10 @@ The members of a group reinitialize it using the following steps:
     * The `group_id`, `version`, and `cipher_suite` fields in the Welcome
       message MUST be the same as the corresponding fields in the ReInit
       proposal.
-    * The `epoch` in the Welcome message MUST be 1
+    * The `epoch` in the Welcome message MUST be 1.
     * The Welcome MUST specify a PreSharedKey of type `resumption` with usage
       `reinit`.  The `group_id` must match the old group, and the `epoch` must
       indicate the epoch after the Commit covering the ReInit.
```
Comment (Bren2010) [inline draft-ietf-mls-protocol.md:3186]: The above line about "epoch must be 1" conflicts with this line that "epoch must be +1 the last Commit"

File: draft-ietf-mls-protocol.md [hunk 3]
```diff
@@ -3178,13 +3180,10 @@ The members of a group reinitialize it using the following steps:
     * The `group_id`, `version`, and `cipher_suite` fields in the Welcome
       message MUST be the same as the corresponding fields in the ReInit
       proposal.
-    * The `epoch` in the Welcome message MUST be 1
+    * The `epoch` in the Welcome message MUST be 1.
```
Comment (bifurcation) [inline draft-ietf-mls-protocol.md:3183]:
```suggestion
```

File: draft-ietf-mls-protocol.md (hunk 2 above)
Comment (bifurcation) [reply draft-ietf-mls-protocol.md:3186]: Good point, reverted.

**Comment by bifurcation** [comment]
Interim 2022-05-26:

* Ready to go once merge conflicts are resolved.

**Pull request merged**